### ISSUE TITLE
@Route("") is not rendered if views/$layout.tsx is present

### ISSUE THREAD

**Issue by Legioth** [vaadin/flow#18966]
### Description of the bug

A Flow view with `@Route("")` is not rendered if the application has a client-side main layout (`views/$layout.tsx`). Flow views with other routes are rendered and the `""` view is also rendered if `$layout.tsx` is removed.

### Expected behavior

The view should render

### Minimal reproducible example

1. Get a starter with Vaadin 24.4.0.alpha14 and remove all Flow and/or Hilla views
2. Add a simple Flow view with `@Route("")`
3. Add a simple Hilla view in `views/hilla.tsx`
4. Launch and see that you can navigate to both views
5. Add a simple `views/$layout.tsx` file.
6. Observe that the Flow view no longer renders
7. Change the Flow view to e.g. `@Route("flow")` and notice that it renders again
8. Change back to `@Route("")` and disable the layout by renaming it to `_$layout.tsx`. Notice that the view renders again.

You can use this simple content for `$layout.tsx`
```
import { Outlet } from 'react-router-dom';

export default function MainLayout() {
    return <Outlet />
}
```

### Versions

- Vaadin / Flow version: Vaadin 24.4.0.alpha14
- Java version: 21

**Comment by caalador** [comment]
So it seems that Hilla file routes always adds the `$layout.js` with the path `""` even though as a general layout it shouldn't have a path to it at all. So as this is registered to / then the Flow route / is not resolved as going for / will render the $layout and not flow which is under `/*`.

This actually caused another issue in the routing when instead of getting only match for `/*` we got a match for `/`and `/*` which was unexpected.

**Comment by caalador** [comment]
Will wait for the result from vaadin/hilla#2238 if that handles the layout route correctly

**Comment by caalador** [comment]
Right now running with 24.4.-SNAPSHOT just having a hilla view without a defined `@layout.js` will generate `createRoute("", [createRoute("hilla", Page0)]);` which overrides the server  / route removing the first `createRoute` to get `createRoute("hilla", Page0)` makes flow / route render as expected.

**Comment by mshabarov** [comment]
Just tested as well with the 24.4.0.alpha21 and noticed no changes, since last testing: if I add `AtRoute("")` Flow view, having `@layout.tsx`, then I get a blank page when navigate to `/`. Changing the mapping to `/foo` makes the content rendered.
Thus, the issue is still valid.

**Comment by mshabarov** [comment]
Okay, it turns into even more severe issue: if you simply have Flow view with `AtRoute("")` and any Hilla view (without layout), then you will see a blank page when you navigate to `/`.

**Comment by caalador** [comment]
Issue seems to have been fixed in vaadin/hilla#2315
